docs/codeql/codeql-for-visual-studio-code/running-codeql-queries-at-scale-with-mrva.rst

```diff
@@ -0,0 +1,184 @@
+
+   - Select **Top X repositories**.
+   - Click **Add new database**, the **+** icon, to select either a single repository, or all repositories in an organization.
+   - Click **LIST-NAME** to select a custom list that you have created previously.
+   - Click **Add new list**, the folder icon, to create a custom list (see later in this article for details).
+
+        .. image:: ../images/codeql-for-visual-studio-code/variant-analysis-repo-lists.png
+            :width: 350
+            :alt: Screenshot of the CodeQL extension in Visual Studio Code. The "Variant Analysis Repositories" section is expanded and the header buttons are highlighted with a dark orange outline. The "Top 10 repositories" item has a checkmark to show that it is currently selected.
+
+#. Open the query you want to run, right-click in the query file, and select **CodeQL: Run Variant Analysis** to start variant analysis.
+
+The CodeQL extension builds a CodeQL pack with your library and any library dependencies. The CodeQL pack and your selected repository list are posted to an API endpoint on GitHub.com which triggers a GitHub Actions dynamic workflow in your controller repository. The workflow spins up multiple parallel jobs to execute the CodeQL query against the repositories in the list, optimizing query execution. As each workflow run finishes, the results are processed and displayed in a variant analysis results view in Visual Studio Code.
+
+.. pull-quote::
+
+    Note
+
+    If you need to cancel the variant analysis run for any reason, click **Stop query** in the Variant Analysis Results view.
+
+Exploring your results
+----------------------
+
+When you run variant analysis, as soon as a workflow to run your analysis on GitHub is running, a Variant Analysis Results view opens to display the results as soon as they are ready. You can use this view to monitor progress, see any errors, and access the workflow logs in your controller repository.
+
+.. image:: ../images/codeql-for-visual-studio-code/variant-analysis-results-view.png
+    :alt: Screenshot of the "Variant Analysis Results" view showing a partially complete run. Analysis of ``angular/angular`` is still running but all other results are displayed. ``facebook/create-react-app`` has three results for this query.
+
+When your variant analysis run is scheduled, the results view automatically opens. Initially the view shows a list of every repository that was scheduled for analysis. As each repository is analyzed, the view is updated to show a summary of the number of results. To view the detailed results for a repository (including results paths), click the repository name.
+
+For each repository, you can see:
+
+- Number of results found by the query
+- Visibility of the repository
+- Whether analysis is still running (black, moving circle) or finished (green checkmark)
+- Number of stars the repository has on GitHub
+- How long ago the CodeQL database that was analyzed was created
```

Comment:
This is actually the time the repo was last updated (which can be various different events, not just a commit). We've had several goes at choosing the wording and there is always this confusion 😬 

```suggestion
- How long ago the repository was last updated
```

Reply:
🤔 - so the repository could have been updated 5 minutes ago, but the CodeQL database might be a week old because analysis is broken? Or do we only analyze repositories where the CodeQL database is reasonably up to date for the codebase?

Reply:
>so the repository could have been updated 5 minutes ago, but the CodeQL database might be a week old because analysis is broken? Or do we only analyze repositories where the CodeQL database is reasonably up to date for the codebase?

That's right. We have no clever ways of filtering repos based on when they were last updated.

Those metrics (stars, last updated) give the researcher some insight into the repos they are targeting: high star count = popular so impactful if a bug is found. Not updated recently = outdated and possibly vulnerable code.